### PR TITLE
1.3/1043 create a voucher allocation page

### PR DESCRIPTION
**Pull request by alexalala**
Ignore the gross css file, I'd set fire to it and start again if I could 😋 

Kat's had a look at it and is really happy with it for the session on Friday. It's unfinished, but will be finished in the sprint in the following two weeks.

https://trello.com/c/sMRrAAv7/1043-create-a-voucher-allocation-page-ee-vvv

### CODE REVIEW

**Comment by katjam** [comment]
@alexalala so this needs to be merged and pushed to staging for a testing session?

File: public/store/css/main.css [hunk 1]
```diff
@@ -589,6 +593,7 @@ EDIT PAGE
 
 .clickable-span {
 	cursor: pointer;
+  text-decoration: underline;
```
Comment (katjam) [inline public/store/css/main.css]: lol gross... just kidding - but you need couple more spaces to get the indent matching.

File: public/store/css/main.css [hunk 2]
```diff
@@ -728,3 +733,84 @@ and (max-device-width : 1024px)  {
...
+
+*/
+
+.alongside-container {
+  display: flex;
+  align-items: flex-end;
+  justify-content: space-between;
+}
+
+.alongside-container > div,
```
Comment (katjam) [inline public/store/css/main.css]: Loving the layout by %% :)

File: public/store/js/slideout.min.js [hunk 3]
```diff
@@ -0,0 +1 @@
...!s){i(u);s=true}}function u(){n.call(t,o);s=false}t.addEventListener(e,r,false);return r}e.exports=o},{}],3:[function(t,e,n){"use strict";var i=function(t,e){if(!(t instanceof e)){throw new TypeError("Cannot call a class as a function")}};n.__esModule=true;var o=function(){function t(){i(this,t)}t.prototype.on=function e(t,n){this._eventCollection=this._eventCollection||{};this._eventCollection[t]=this._eventCollection[t]||[];this._eventCollection[t].push(n);return this};t.prototype.once=function n(t,e){var n=this;function i(){n.off(t,i);e.apply(this,arguments)}i.listener=e;this.on(t,i);return this};t.prototype.off=function o(t,e){var n=undefined;if(!this._eventCollection||!(n=this._eventCollection[t])){return this}n.forEach(function(t,i){if(t===e||t.listener===e){n.splice(i,1)}});if(n.length===0){delete this._eventCollection[t]}return this};t.prototype.emit=function s(t){var e=this;for(var n=arguments.length,i=Array(n>1?n-1:0),o=1;o<n;o++){i[o-1]=arguments[o]}var s=undefined;if(!this._eventCollection||!(s=this._eventCollection[t])){return this}s=s.slice(0);s.forEach(function(t){return t.apply(e,i)});return this};return t}();n["default"]=o;e.exports=n["default"]},{}]},{},[1])(1)});
```
Comment (katjam) [inline public/store/js/slideout.min.js]: Assuming this is a trusted library? Is there a better way to import it?

Comment (katjam) [reply public/store/js/slideout.min.js]: I'm guessing not set up with any form of node package manager?

File: resources/views/store/voucher_allocation.blade.php [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    @include('store.partials.navbar', ['headerTitle' => 'Voucher Collection'])
+
+    <div class="content">
+        <form>
+            <div class="col">
+                <div>
+                    <img src="{{ asset('store/assets/info-light.svg') }}">
+                    <h2>This Family</h2>
+                </div>
+                <div class="alongside-container">
```
Comment (katjam) [inline resources/views/store/voucher_allocation.blade.php]: This is intentionally hardcoded for the testing session?

Has it been carded to fill with dynamic values? So we don't forget.

File: resources/views/store/voucher_allocation.blade.php [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+                        </ul>
+                    </div>
+                </div>
+                <div class="warning">
+                    <p class="v-spaced">
+                        <i class="fa fa-exclamation-circle" aria-hidden="true"></i>
+                        Next month a child has a birthday, so the voucher allocation will change.
+                    </p>
+                </div>
+                <button class="short-button">Go to edit family</button>
```
Comment (katjam) [inline resources/views/store/voucher_allocation.blade.php]: Not wired up - intentional?

File: resources/views/store/voucher_allocation.blade.php [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+                    <label for="add-voucher-input">Add individual vouchers</label>
+                    <div class="small-button-container">
+                        <input type="text" id="add-voucher-input">
+                        <button class="addButton">
+                            <i class="fa fa-plus" aria-hidden="true"></i>
+                        </button>
+                    </div>
+                </div>
+                <p class="center vh-spaced">You have added # vouchers</p>
+                <button id="collection-button" class="long-button">Go to voucher collection</button>
```
Comment (katjam) [inline resources/views/store/voucher_allocation.blade.php]: So this is for faking the transaction?

File: public/store/js/slideout.min.js (hunk 3 above)
Comment (alexalala) [reply public/store/js/slideout.min.js]: Good spot, this is old!

**Pull request merged**